Unsupported platforms should fail earlier in the configuration process

When trying to cross-compile, e.g. to web assembly, and I configure like 

```
emconfigure ./configure --host=wasm32-unknown-emscripten 
```

and then try to build it using

```
emmake make
```

The compilation fails with

```
  CC       libcrypto_la-cpt_err.lo
In file included from cpt_err.c:58:
In file included from /usr/local/include/openssl/opensslconf.h:15:
/usr/local/include/openssl/macros.h:147:4: error: "OPENSSL_API_COMPAT expresses an impossible API compatibility level"
  147 | #  error "OPENSSL_API_COMPAT expresses an impossible API compatibility level"
      |    ^
1 error generated.
```

But why on earth would the code even read a header file from the OpenSSL/LibreSSL library I currently have installed?  I am trying to build this stuff right now. My system should not even be required to have OpenSSL/LibreSSL installed, otherwise it creates a chicken-egg problem. IMHO, the source code should never reference any OpenSSL/LibreSSL header files in my current system or build environment, only from the source directory of the code I'm trying to build.

It even makes less sense since I'm cross compiling, so my locally installed headers must be totally irrelevant as my local system is not the target system the code will be running on.

If I remove that file, the build fails as well

```
/Library/Automount/Internal/Xcode/App/Xcode_15.3.app/Contents/Developer/usr/bin/make  all-am
  CC       libcrypto_la-cpt_err.lo
cpt_err.c:58:10: fatal error: 'openssl/opensslconf.h' file not found
   58 | #include <openssl/opensslconf.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

so you cannot cross-build LibreSSL unless you have either LibreSSL or OpenSSL installed on your build system?

--- a/configure.ac
+++ b/configure.ac
@@ -32,6 +32,10 @@ LT_INIT([pic-only])
 
 CHECK_OS_OPTIONS
 
+if test "$HOST_OS" = "unsupported"; then
+	AC_MSG_ERROR([unsupported platform: $host_os])
+fi
+
 CHECK_C_HARDENING_OPTIONS
 
 DISABLE_AS_EXECUTABLE_STACK

--- a/m4/check-os-options.m4
+++ b/m4/check-os-options.m4
@@ -109,7 +109,7 @@ char buf[1]; getentropy(buf, 1);
 		)
 		CPPFLAGS="$CPPFLAGS -D_OPENBSD_SOURCE"
 		;;
-	*openbsd* | *bitrig*)
+	*openbsd*)
 		HOST_OS=openbsd
 		HOST_ABI=elf
 		AC_DEFINE([HAVE_ATTRIBUTE__BOUNDED__], [1], [OpenBSD gcc has bounded])
@@ -131,7 +131,9 @@ char buf[1]; getentropy(buf, 1);
 		CPPFLAGS="$CPPFLAGS -D__EXTENSIONS__ -D_XOPEN_SOURCE=600 -DBSD_COMP"
 		AC_SUBST([PLATFORM_LDADD], ['-ldl -lmd -lnsl -lsocket'])
 		;;
-	*) ;;
+	*)
+		HOST_OS=unsupported
+		;;
 esac
 
 # Check if time_t is sized correctly

